sdk/python/feast/registry.py

```diff
@@ -480,12 +548,34 @@ def get_on_demand_feature_view(
                 return OnDemandFeatureView.from_proto(on_demand_feature_view)
         raise OnDemandFeatureViewNotFoundException(name, project=project)
 
+    def get_data_source(
+        self, name: str, project: str, allow_cache: bool = False
+    ) -> DataSource:
+        """
+        Retrieves a data source.
+
+        Args:
+            name: Name of data source
+            project: Feast project that this data source belongs to
+            allow_cache: Whether to allow returning this data source from a cached registry
+
+        Returns:
+            Returns either the specified data source, or raises an exception if none is found
+        """
+        registry = self._get_registry_proto(allow_cache=allow_cache)
+
+        for data_source in registry.data_sources:
+            if data_source.project == project and data_source.name == name:
+                return DataSource.from_proto(data_source)
+        raise DataSourceObjectNotFoundException(name, project=project)
+
     def apply_materialization(
         self,
         feature_view: FeatureView,
         project: str,
         start_date: datetime,
         end_date: datetime,
+        earliest_latest_event_timestamp: Optional[Tuple[datetime, datetime]] = None,
```

Comment:
why is this being added?

Reply:
Removed. An artifact that I missed from a larger PR